src/util/http/websocket/WebSocketSession.cpp

```diff
@@ -28,14 +29,33 @@ std::string extractQueryId(std::string_view path) {
 
 // _____________________________________________________________________________
 
+bool WebSocketSession::tryToCancelQuery() const {
+  if (auto cancellationHandle =
+          queryRegistry_.getCancellationHandle(queryId_)) {
+    cancellationHandle->cancel(CancellationState::MANUAL);
+
+    return true;
+  }
+  return false;
+}
+
+// _____________________________________________________________________________
+
 net::awaitable<void> WebSocketSession::handleClientCommands() {
   beast::flat_buffer buffer;
 
   while (ws_.is_open()) {
     co_await ws_.async_read(buffer, net::use_awaitable);
-    // TODO<RobinTF> replace with cancellation process
-    ws_.text(ws_.got_text());
-    co_await ws_.async_write(buffer.data(), net::use_awaitable);
+    if (ws_.got_text()) {
+      auto data = buffer.data();
+      std::string_view dataAsString{static_cast<char*>(data.data()),
+                                    data.size()};
+      if (dataAsString == "cancel_on_close") {
+        cancelOnClose_ = true;
+      } else if (dataAsString == "cancel" && tryToCancelQuery()) {
+        break;
```

Comment:
Currently there's no real protocol there, the server just sends JSON that is always treated as serialized runtime information. But with enough effort this could be changed of course.